wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -261,26 +276,316 @@ func (v *Wrapper) Decrypt(ctx context.Context, in *wrapping.EncryptedBlobInfo, a
 	return wrapping.NewEnvelope(nil).Decrypt(envInfo, aad)
 }
 
-func (v *Wrapper) ImportKey(_ context.Context, _ string, _ wrapping.KMSKey) (string, error) {
-	return "", nil
+// ImportKey imports the given key into Azure Key Vault by implementing the
+// Azure Key Vault bring your own key (BYOK) specification. The BYOK specification is
+// detailed at https://docs.microsoft.com/en-us/azure/key-vault/keys/byok-specification.
+func (v *Wrapper) ImportKey(ctx context.Context, name string, key wrapping.KMSKey) (string, error) {
+	if err := v.validateKMSKey(key); err != nil {
+		return "", err
+	}
+
+	// Generate a UUID for the name of the Key Exchange Key (KEK)
+	kekName, err := uuid.GenerateUUID()
+	if err != nil {
+		return "", err
+	}
+
+	// Generate an HSM backed RSA key pair in Azure Key Vault.
+	// This key will be used as the KEK and has an expiration.
+	exp := date.UnixTime(time.Now().Add(3 * time.Minute))
+	kekBundle, err := v.client.CreateKey(ctx, v.baseURL, kekName, keyvault.KeyCreateParameters{
+		Kty:     keyvault.RSAHSM,
+		KeySize: to.Int32Ptr(2048),
+		KeyOps:  &[]keyvault.JSONWebKeyOperation{"import"},
+		KeyAttributes: &keyvault.KeyAttributes{
+			Expires: &exp,
+		},
+	})
+	if err != nil || kekBundle.Key == nil {
+		return "", fmt.Errorf("error generating KEK: %w", err)
+	}
+	defer func() {
+		if _, err := v.client.DeleteKey(ctx, v.baseURL, kekName); err != nil {
+			v.logger.Warn("error deleting KEK", "name", kekName, "error", err)
+		}
+	}()
+
+	// Parse the RSA public key of the KEK
+	kekPubKey, err := jwkToRSAPublicKey(kekBundle.Key)
+	if err != nil {
+		return "", err
+	}
+
+	// Produce the target key plaintext
+	var targetKey []byte
+	switch key.Type {
+	case wrapping.RSA2048, wrapping.RSA3072, wrapping.RSA4096:
+		// For an RSA key, the private key ASN.1 DER encoding [RFC3447] wrapped in PKCS#8 [RFC5208]
+		if targetKey, err = x509.MarshalPKCS8PrivateKey(key.Material.RSAKey); err != nil {
+			return "", err
+		}
```

Comment:
Should we guard against unsupported key types via the default case?

Reply:
Good idea. I've added a default case in 8516620.